File: web-client/chinese-chess/src/scene/play/ChessAction.ts
import DisplayChess from "./DisplayChess";
import Chess from "./rule/Chess";
import ChessPos from "./rule/ChessPos";
import ChessHost from "./rule/chess_host";

export default class ChessAction {
    chessHost: ChessHost;
    chessType: Function;
    fromPos: ChessPos;
    toPos: ChessPos;
    eatenChess: DisplayChess | Chess;
}

File: server/src/main/java/io/github/hulang1024/chinesechess/play/ChessAction.java
package io.github.hulang1024.chinesechess.play;

import io.github.hulang1024.chinesechess.play.rule.Chess;
import io.github.hulang1024.chinesechess.play.rule.ChessEnum;
import io.github.hulang1024.chinesechess.play.rule.ChessHost;
import io.github.hulang1024.chinesechess.play.rule.ChessPos;
import lombok.Data;

@Data
public class ChessAction {
    private ChessHost chessHost;
    private ChessEnum chessType;
    private ChessPos fromPos;
    private ChessPos toPos;
    private Chess eatenChess;
}
